### PR TITLE
front: fix selected color, typos, and add warning in speed-effort curve editor

### LINKED ISSUE #6843: Front: Rolling editor have the wrong color selection for speed-effort curves
### What happened?

When we try to create or edit a rolling stock, in the speed effort curves page, when we add some data, the selection color for traction modes, electrical profiles and power restriction are red, it should be the same as comfort levels.

test4

### What did you expect to happen?

To have the right color

### How can we reproduce it (as minimally and precisely as possible)?

1. go to the rolling stock editor
2. try to edit speed-effort curves

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Windows 11
- **Env**: Dev dgex

### OSRD version (top right corner `Account` button > `Informations`)

dev 02cebc2

### LINKED ISSUE #6859: front: notify the user when there no power class in the rolling stock editor
### Description and goal

In the rolling stock editor, when we try to edit speed-effort curves, we could save it without sending any power class matching the power restriction.
test5

in this case, there is an empty string sent to the back, and the simulation will fallback on unspecified curves (to keep it working).
However this behavior is not recommanded, so we should warn the user about it.

### Acceptance criteria

if there is no power class associated to the power restriction :
- [ ] the form is circle in yellow
- [ ] there is a warning message for the user
- [ ] the user could still save it empty if he wants to

### LINKED ISSUE #6912: Missing translation in rolling stock editor when deleting an object
### What happened?

Capture d’écran 2024-03-15 à 11 35 40

### What did you expect to happen?

Have a better text

### How can we reproduce it (as minimally and precisely as possible)?

Go to Rolling stock editor, try to delete a power restriction & power class or an electrical profile

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

dev b5e1083

### PR DESCRIPTION
**Pull request by Akctarus**
closes #6843 
closes #6859 
closes #6912

### CODE REVIEW

File: front/src/common/Selector/Selector.tsx [hunk 1]
```diff
@@ -148,7 +156,7 @@ const Selector = <
   }>
 ) => {
   const {
-    borderClass = 'selector-pink',
+    borderClass = 'selector-blue',
```
Comment (clarani) [inline front/src/common/Selector/Selector.tsx:158]: Can you also remove the borderClass in CurveParamSelectors l257 please ? :)

File: front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx [hunk 2]
```diff
@@ -338,6 +341,16 @@ const CurveParamSelectors = ({
               }}
               dataTestId="power-restriction-selector"
             />
+            {isWarning && (
+              <div className="warning-message text-secondary">
+                <div className="pl-3 d-flex align-items-center">
+                  <div className="mr-2 warning-icon">
+                    <Alert />
+                  </div>
+                  <span>{t('missingPowerClass')}</span>
```
Comment (clarani) [inline front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx]: You can even pluralize the message: you just need to add a param count in t and in the translation file, you need to add 2 translations `missingPowerClass_one` and `missingPowerClass_other`
```suggestion
                  <span>{t('missingPowerClass', {count: missingPowerClassesNb})}</span>
```

File: front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss [hunk 3]
```diff
@@ -295,4 +295,13 @@
...
     }
   }
+
+  .warning-message {
+    border: 2px solid var(--yellow);
+    border-radius: 0.3rem;
+  }
+
+  .warning-icon {
+    color: var(--yellow);
```
Comment (clarani) [inline front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss]: You can move this class in the warning-message class (I think)
```suggestion
    .warning-icon {
      color: var(--yellow);
    }
```

File: front/src/styles/scss/common/components/_selector.scss [hunk 4]
```diff
@@ -44,16 +49,12 @@
     }
     &.selected {
       &.selector-blue {
```
Comment (clarani) [inline front/src/styles/scss/common/components/_selector.scss]: You can even remove this class I think
```suggestion
```

File: front/src/styles/scss/common/components/_selector.scss [hunk 5]
```diff
@@ -44,16 +49,12 @@
...
-        outline: 0.13rem solid var(--blue);
-      }
-      &.selector-pink {
-        outline: 0.13rem solid #fd2f7e;
+        box-shadow: inset 0 0 0 0.13rem var(--blue);
       }
     }
     &.hovered {
-      outline: 0.13rem solid #fd2f7e80;
       &.selector-blue {
```
Comment (clarani) [inline front/src/styles/scss/common/components/_selector.scss]: Same here, you can remove it
```suggestion
```

File: front/src/common/BootstrapSNCF/InputSNCF.tsx [hunk 6]
```diff
@@ -19,6 +19,7 @@ export type InputSNCFProps = {
   max?: string | number;
   list?: string;
   // Error handling
+  isWarning?: boolean;
```
Comment (Uriel-Sautron) [inline front/src/common/BootstrapSNCF/InputSNCF.tsx]: isWarning is declared but never used

**Review by clarani — COMMENTED** [review]
Very nice PR, thanks for all this fixes :)

**Review by Uriel-Sautron — COMMENTED** [review]
Good job, thank you for this PR. Tested on local.

File: front/src/common/BootstrapSNCF/InputSNCF.tsx (hunk 6 above)
Comment (Akctarus) [reply front/src/common/BootstrapSNCF/InputSNCF.tsx]: TY, fixed

File: front/src/common/Selector/Selector.tsx (hunk 1 above)
Comment (Akctarus) [reply front/src/common/Selector/Selector.tsx:158]: fixed

File: front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss (hunk 3 above)
Comment (Akctarus) [reply front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss]: fixed

File: front/src/styles/scss/common/components/_selector.scss (hunk 4 above)
Comment (Akctarus) [reply front/src/styles/scss/common/components/_selector.scss]: fixed

File: front/src/styles/scss/common/components/_selector.scss (hunk 5 above)
Comment (Akctarus) [reply front/src/styles/scss/common/components/_selector.scss]: fixed

File: front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx (hunk 2 above)
Comment (Akctarus) [reply front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx]: Fixed, ty for the tip !

File: front/src/styles/scss/common/components/_selector.scss
Comment (clarani) [inline front/src/styles/scss/common/components/_selector.scss:1]: I think you can lint this file (there is 2 extra spaces before box-shadow l51 and 54

File: front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx [hunk 7]
```diff
@@ -244,7 +244,10 @@ const CurveParamSelectors = ({
     );
   };
 
-  const isWarning = Object.values(extraColumnData.data).includes('');
+  const isWarning = Object.values(extraColumnData.data).reduce(
```
Comment (clarani) [inline front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx]: warningsNb or warningsCount ?
```suggestion
  const warningsNb = Object.values(extraColumnData.data).reduce(
```

**Review by clarani — APPROVED** [review]
LGTM, you can fix my last 2 comments, squash and merge :+1:

File: front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx (hunk 7 above)
Comment (Akctarus) [reply front/src/modules/rollingStock/components/RollingStockEditor/CurveParamSelectors.tsx]: Fixed, ty for the comments !

**Pull request merged**